src/pkg/utils/template_test.go

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The UDS Authors
+
+package utils
+
+import (
+	"testing"
+
+	zarfTypes "github.com/defenseunicorns/zarf/src/types"
+
+	"github.com/defenseunicorns/maru-runner/src/types"
+	zarfUtils "github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func TestTemplateString(t *testing.T) {
+	type args struct {
+		templateMap map[string]*zarfUtils.TextTemplate
+		s           string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "single replacement",
+			args: args{
+				templateMap: map[string]*zarfUtils.TextTemplate{
+					"${VAR1}": {Value: "replacement1"},
+				},
+				s: "This is a ${VAR1} string.",
+			},
+			want: "This is a replacement1 string.",
+		},
+		{
+			name: "multiple replacements",
+			args: args{
+				templateMap: map[string]*zarfUtils.TextTemplate{
+					"${VAR1}": {Value: "replacement1"},
+					"${VAR2}": {Value: "replacement2"},
+				},
+				s: "This is a ${VAR1} and ${VAR2} string.",
+			},
+			want: "This is a replacement1 and replacement2 string.",
+		},
+		{
+			name: "no replacements",
+			args: args{
+				templateMap: map[string]*zarfUtils.TextTemplate{},
+				s:           "This string has no variables.",
+			},
+			want: "This string has no variables.",
+		},
+		{
+			name: "missing replacement",
+			args: args{
+				templateMap: map[string]*zarfUtils.TextTemplate{
+					"${VAR1}": {Value: "replacement1"},
+				},
+				s: "This is a ${VAR1} and ${MISSING_VAR} string.",
+			},
+			want: "This is a replacement1 and ${MISSING_VAR} string.",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := TemplateString(tt.args.templateMap, tt.args.s); got != tt.want {
```

Comment:
Instead of manually rolling the checks, in other tests we use the `require` function from `testify` which simpliies this logic